src/MetaCompilation/MetaCompilation/MetaCompilation/DiagnosticAnalyzer.cs

```diff
@@ -1302,317 +1294,315 @@ private void IfDiagnostic(CompilationAnalysisContext context, StatementSyntax st
-                                        IfDiagnostic(context, statements[8], DiagnosticIncorrectRule, ruleNames[0], locationToken.Text);
+                                        var argument = ((statements[4] as LocalDeclarationStatementSyntax).Declaration.Variables[0].Initializer.Value as InvocationExpressionSyntax).ArgumentList.Arguments[0];
+                                        ReportDiagnostic(context, StartSpanMissingRule, argument.GetLocation(), locationCorrect, keywordIdentifierToken.Text);
                                         return false;
                                     }
-
-                                    if (statementCount > 9)
-                                    {
-                                        bool reportCorrect = DiagnosticReportCheck(context, diagnosticToken, contextParameter, statements);
-                                        if (!reportCorrect)
-                                        {
-                                            IfDiagnostic(context, statements[9], DiagnosticReportIncorrectRule, contextParameter.Identifier.Text, diagnosticToken.Text);
-                                            return false;
-                                        }
-                                    }
                                     else
                                     {
-                                        ReportDiagnostic(context, DiagnosticReportMissingRule, statements[8].GetLocation(), contextParameter.Identifier.Text);
+                                        var argument = ((statements[5] as LocalDeclarationStatementSyntax).Declaration.Variables[0].Initializer.Value as InvocationExpressionSyntax).ArgumentList.Arguments[0];
+                                        ReportDiagnostic(context, StartSpanMissingRule, argument.GetLocation(), locationCorrect, keywordIdentifierToken.Text);
                                         return false;
                                     }
                                 }
-                                else
+                                
+                                if (startSpanCorrect && endSpanCorrect)
                                 {
-                                    ReportDiagnostic(context, DiagnosticMissingRule, statements[7].GetLocation());
+                                    ReportDiagnostic(context, DiagnosticReportMissingRule, statements[7].GetLocation(), contextParameter.Identifier.Text);
                                     return false;
                                 }
                             }
                             else
                             {
-                                ReportDiagnostic(context, LocationMissingRule, statements[6].GetLocation(), statementIdentifierToken.Text, spanToken.Text);
+                                var argument = ((statements[3] as LocalDeclarationStatementSyntax).Declaration.Variables[0].Initializer.Value as InvocationExpressionSyntax).ArgumentList.Arguments[1];
```

Comment:
@jepetty and again, I think this is squiggling the wrong statement? Should be 4 maybe?